### ISSUE TITLE
stlightly inaccurate plot(".") vs plot("-") positioning

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#7624]
```
plot(range(1000), "."); plot(range(1000))
```
gives
![foo](https://cloud.githubusercontent.com/assets/1322974/21156252/0500fdce-c12a-11e6-9f87-d5d6b170ebcd.png)

Note that the line is clearly drawn a bit higher on average than the dots.

I thought this was a duplicate of #7262 but actually it is not, this issue is present even in 1.3.

**Comment by anntzer** [comment]
(also fixed by mplcairo)

**Comment by jklymak** [comment]
Yes, I agree that there is a one-pixel offset at any resolution.  The line is one pixel higher than the center of the markers, even with `snap=False`.

**Comment by jklymak** [comment]
BTW, not sure if this is out issue or Agg's.  I would tend to guess maybe Agg, but perhaps we do some messing with the marker positions?
